File: templates/Uwp/_composition/MVVMLight/Page.AddBackCommand/ViewModels/wts.ItemNameDetailViewModel_postaction.cs
//{[{
using System.Windows.Input;
using Param_RootNamespace.Helpers;
using Param_RootNamespace.Services;
using GalaSoft.MvvmLight.Command;
//}]}

namespace Param_RootNamespace.ViewModels
{
    public class wts.ItemNameDetailViewModel : ViewModelBase
    {
//^^
//{[{
        public static NavigationServiceEx NavigationService => ViewModelLocator.Current.NavigationService;

        private ICommand _goBackCommand;

        public ICommand GoBackCommand => _goBackCommand ?? (_goBackCommand = new RelayCommand(OnGoBack));
//}]}

        public wts.ItemNameDetailViewModel()
        {
        }

//^^
//{[{
        private void OnGoBack()
        {
            if (NavigationService.CanGoBack)
            {
                NavigationService.GoBack();
            }
        }
//}]}
    }
}

File: templates/Uwp/_composition/MVVMLight/Page.AddBackCommand._VB/ViewModels/wts.ItemNameDetailViewModel_postaction.vb
'{[{
Imports Param_RootNamespace.Services
Imports GalaSoft.MvvmLight.Command
'}]}
Namespace ViewModels
    Public Class wts.ItemNameDetailViewModel
        Inherits ViewModelBase

        '^^
        '{[{

        Private ReadOnly Property NavigationService As NavigationServiceEx
            Get
                Return ViewModelLocator.Current.NavigationService
            End Get
        End Property

        Public ReadOnly Property GoBackCommand As ICommand = new RelayCommand(AddressOf OnGoBack)
        '}]}

        Public Sub New()
        End Sub

        '^^
        '{[{
        Private Sub OnGoBack()
            If NavigationService.CanGoBack Then
                NavigationService.GoBack()
            End If
        End Sub
        '}]}
    End Class
End Namespace
